### ISSUE TITLE
Events not being sent back to ElasticSearch if the connection is restored

### ISSUE THREAD

**Issue by mperciun83** [serilog/serilog-sinks-elasticsearch#252]
**A few questions before you begin:**

> Is this an issue related to the [Serilog core project](https://github.com/serilog/serilog) or one of the [sinks](https://github.com/serilog/serilog/wiki/Provided-Sinks) or 
[community projects](https://github.com/serilog/serilog/wiki/Community-Projects).  
This issue list is intended for Serilog Elasticsearch Sink issues. If this issue relates to another sink or to the code project, 
please log on the related repository.  Please use [Gitter chat](https://gitter.im/serilog/serilog) and [Stack Overflow](http://stackoverflow.com/questions/tagged/serilog)  for discussions and questons.

Yes this issue is related to the ElasticSearch sink.

**Does this issue relate to a new *feature* or an existing *bug*?**
- [x] Bug
- [ ] New Feature

**What version of Serilog.Sinks.Elasticsearch is affected?  Please list the related NuGet package.**
7.1.0

**What is the target framework and operating system? See [target frameworks](https://docs.microsoft.com/en-us/nuget/schema/target-frameworks) & [net standard matrix](https://docs.microsoft.com/en-us/dotnet/standard/net-standard).**

- [x] netCore 2.0
- [ ] netCore 1.0
- [ ] 4.7
- [ ] 4.6.x
- [ ] 4.5.x

**Please describe the current behavior?**
I am facing an issue with the durability feature of this sink.
My application's log events are correctly pushed to an ElasticSearch index. The problem occurs when the index becomes unavailable and then the events written in the bookmark file are not sent back to ElasticSearch after the index becomes available and the application is restarted. If the index is unavailable I am also seeing a lot of invalid-200.json files corresponding to the unsent events.

**Please describe the expected behavior?**
IF there is an issue with the connectivity between the Sink and the ElasticSeach server should not the bookmarked logs be sent once the connectivity is restored?

**If the current behavior is a bug, please provide the steps to reproduce the issue and if possible a minimal demo of the problem**

1. Create a small console .NET Core application and configure Serilog to use the self debug log.
2. Use Autofac as the IOC implementation and add the following Serilog extension: https://github.com/nblumhardt/autofac-serilog-integration
3. Using the following Nugget version for this demo:
- Autofac 4.9.2
- Serilog 2.8.1-dev-01054
- Serilog.Sinks.ElasticSearch 7.1.0
- Serilog.Settings.Configuration 3.1.0
- Serilog.Sinks.File 4.0.0
3. Run the console app and see the logs being pushed correctly to all configured sinks. The buffer file is created correctly. Close the console
4. Make the index read only
5. Re-run the application and see the logs being pushed correctly to the Console and the File sinks but no to elastic search. The serilog self debug log has a lot of the following errors:

2019-06-21T13:48:33.5349452Z Received failed ElasticSearch shipping result 403: {"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}. Failed payload : {"@timestamp":"2019-06-21T16:48:31.8932771+03:00","level":"Information","messageTemplate":"This is an audit message with {@AuditUserDto}","message":"This is an audit message with AuditDataDto { UserId: 2, TenantId: 1, TenantUserId: null, CorrelationId: 225a3656-b66d-44a0-874d-234dcb03eb30, ChannelId: null, BotId: null, ConnectorHostId: null, Message: \"This is step 0 of execution\", Exception: null, FlowId: null, FlowName: null, FlowStepId: null, FlowStepName: null, ConnectorActionId: null, ConnectorActionName: null, ConnectorTaskId: null, ConnectorTaskName: null }","AuditUserDto":{"_typeTag":"AuditDataDto","UserId":2,"TenantId":1,"TenantUserId":null,"CorrelationId":"225a3656-b66d-44a0-874d-234dcb03eb30","ChannelId":null,"BotId":null,"ConnectorHostId":null,"Message":"This is step 0 of execution","Exception":null,"FlowId":null,"FlowName":null,"FlowStepId":null,"FlowStepName":null,"ConnectorActionId":null,"ConnectorActionName":null,"ConnectorTaskId":null,"ConnectorTaskName":null},"SourceContext":"SerilogConsole.LogWriter","MachineName":"\"DESKTOP-HSAMU3J\"","ThreadId":1,"CustomProperty":"72ba1f99-ba98-492d-964c-6028c118d6af","Application":"SerilogConsole.Marius.Perciun"}.

4.  Below I have attached my serilog configuration which resides under an appSettings.json file:
{
  "Serilog": {
    "MinimumLevel": "Debug",
    "WriteTo": [
      {
        "Name": "File",
        "Args": {
          "path": "../../../Logs/SerilogConsole-Log-.json",
          "rollingInterval": "Day",
          "outputTemplate": "{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz} [{Level}] {Message} {Properties:j}{NewLine}{Exception}",
          "formatter": "Serilog.Formatting.Compact.CompactJsonFormatter, Serilog.Formatting.Compact",
          "fileSizeLimitBytes": 104857600

        }
      },
      {
        "Name": "Elasticsearch",
        "Args": {
          "nodeUris": "xxxxxxxxxxxxxxxxxxxxxxxxx",
          "batchPostingLimit": 1,
          "period": 1,
          "inlineFields": true,
          "bufferBaseFilename": "../../../Logs/SerilogConsole-Provisioning-ELK-Buffer",
          "deadLetterIndexName": "dev-deadletter-{0:yyyy.MM}",
          "indexFormat": "dev-serilogpoc-{0:yyyy.MM}",
          "bufferFileSizeLimitBytes": 5242880,
          "bufferLogShippingInterval": 1000,
          "bufferRetainedInvalidPayloadsLimitBytes": 5000,
          "bufferFileCountLimit": 31,
          "connectionGlobalHeaders": "xxxxxxxxxxxxxxxxxxxxx",
          "connectionTimeout": 5,
          "emitEventFailure": "WriteToSelfLog",
          "queueSizeLimit": "100000",
          "autoRegisterTemplate": true,
          "autoRegisterTemplateVersion": "ESv6",
          "overwriteTemplate": false,
          "numberOfShards": 5,
          "numberOfReplicas": 1
        }
      }
    ],
    "Enrich": [ "FromLogContext", "WithMachineName", "WithThreadId", "WithCustomEnricher" ],
    "Properties": {
      "Application": "SerilogConsole.Marius.Perciun"
    }
  }
}

**Comment by VicXue** [comment]
It sounds like I have the same issue. I can successfully create an index, but the transmission got dropped somehow after a short period of time. All logging data got stored into the buffer file. If I keep interacting with my API, sometimes the hit counts of the elasticsearch get increased. However, it seems the new logging data get merged into old entries. I wonder what caused this issue. 
![77hits](https://user-images.githubusercontent.com/19631188/63657509-0ff3db80-c7f7-11e9-9b84-f3b0f8432ffe.JPG)
![127hits](https://user-images.githubusercontent.com/19631188/63657510-0ff3db80-c7f7-11e9-8fb4-0fd43c1cc61f.JPG)

**Comment by kilroyFR** [comment]
Got same issue with .net framework version of serilog elasticsearch sinks.

**Comment by liftyourgame** [comment]
Getting same issue. Logs fine to files. Going to use File Beat for now to get logs to Elastic

**Comment by mivano** [comment]
Old issue, cleaning up. Still facing this issue, feel free to reopen.

**Comment by DronVelikii** [comment]
@mivano I have the same issue as the author with the last version - Serilog.Sinks.ElasticSearch 8.4.1
